src/instructlab/model/download.py

```diff
@@ -12,6 +16,197 @@
+                local_dir=self.download_dest,
+            )
+
+        except Exception as exc:
+            click.secho(
+                f"Downloading GGUF model failed with the following HuggingFace Hub error: {exc}",
+                fg="red",
+            )
+            raise click.exceptions.Exit(1)
+
+    def download_safetensors(self) -> None:
+        try:
+            if not os.path.exists(os.path.join(self.download_dest, self.repository)):
+                os.makedirs(
+                    name=os.path.join(self.download_dest, self.repository),
+                    exist_ok=True,
+                )
+            snapshot_download(
+                token=self.hf_token,
+                repo_id=self.repository,
+                revision=self.release,
+                local_dir=os.path.join(self.download_dest, self.repository),
+            )
+        except Exception as exc:
+            click.secho(
+                f"Downloading safetensors model failed with the following HuggingFace Hub error: {exc}",
+                fg="red",
+            )
+            raise click.exceptions.Exit(1)
+
+
+class OCIDownloader(Downloader):
+    """Class to handle downloading safetensors and GGUF models from Huggingface"""
+
+    def __init__(
+        self,
+        repository: str,
+        release: str,
+        download_dest: str,
+    ) -> None:
+        super().__init__(
+            repository=repository, release=release, download_dest=download_dest
+        )
+        self.repository = repository
+        self.release = release
+        self.download_dest = download_dest
+
+    def _build_oci_model_file_map(self, oci_model_path: str) -> dict:
+        """
+        Helper function to build a mapping between blob files and what they represent
+        """
+        with open(f"{oci_model_path}/index.json", mode="r", encoding="UTF-8") as f:
+            index_ref = json.load(f)
+        index_hash = re.search(
+            "sha256:(.*)", index_ref["manifests"][0]["digest"]
+        ).group(1)
+
+        with open(
+            f"{oci_model_path}/blobs/sha256/{index_hash}", mode="r", encoding="UTF-8"
+        ) as f:
+            index = json.load(f)
+
+        title_ref = "org.opencontainers.image.title"
+        oci_model_file_map = {}
+
+        for layer in index["layers"]:
+            oci_model_file_map[re.search("sha256:(.*)", layer["digest"]).group(1)] = (
+                layer["annotations"][title_ref]
+            )
+
+        return oci_model_file_map
+
+    def download(self):
+        click.echo(
+            f"Downloading model from OCI registry: {self.repository}@{self.release} to {self.download_dest}..."
+        )
+
+        os.makedirs(self.download_dest, exist_ok=True)
+        model_name = self.repository.split("/")[-1]
+        oci_dir = f"{DEFAULTS.OCI_DIR}/{model_name}"
+        os.makedirs(oci_dir, exist_ok=True)
+
+        try:
+            subprocess.run(
+                [
+                    "skopeo",
```

Comment:
A few thoughts:

* This expects that `skopeo` is present on the system. We should validate that upfront.
* Why Skopeo and not `crane`?
* Is skopeo taking care of the checksumming? Or is it done via a separate command?
* Should we support other tools?

Reply:
`crane` does not require a daemon either. Thanks for the clarification!